### ISSUE TITLE
[Suggestion] Customizable Sandwich Bases

### ISSUE THREAD

**Issue by BobVarioa** [ochotonida/some-assembly-required#31]
Before we start, I'm not talking about the already existing method to add other items that function as bread, I'm more suggesting a "sandwich" type system, if that makes sense (which is why there's more to this suggestion :P).

My idea is that there could be a similar format to registering ingredients for registering bread types. This format would include the specifications of the sandwich it was meant to create, for example, if you wanted to create a salad bowl "sandwich" you could only allow it to create open faced sandwiches, whitelist / blacklist ingredients, and perhaps modifiers to the sandwiches effects and/or nutrition. 

Also, in the future you could extend this idea:
- multiple "bites" per sandwich type
- a liquid only "sandwich" (smoothie?)
- different top and bottom bread types (i.e. burger buns)
- limits on the number of ingredients per sandwich type instead of globally

**Comment by ochotonida** [comment]
I'll have to think about this a bit. Some of these features might be a bit out of scope, or difficult to implement in a data pack format while not making it excessively complex but also not too limiting in what you can do with it. I do like some of the ideas though.

I've actually thought about adding burger buns before :p. The main concern I have with those is whether I can make them look good and that they don't feel out of place.

**Comment by ochotonida** [comment]
The burger buns were added a while ago, but I think the remaining suggestions don't really fit with my vision of the mod.

I'm personally not really interested in adding new foods like salads or smoothies, and implementing a data pack format for this doesn't sound feasible.
I like the idea of having multiple bites per sandwich, but the compatibility issues this would cause just wouldn't make it worth it.
